engine/src/engine.rs

```diff
@@ -0,0 +1,146 @@
+    0.75, 0.75, 0.50, -0.50, -0.50, 0.25, 0.00, 0.00, 0.00, 0.00, 0.25, -0.50, -1.00, -0.50, -0.50,
+    -0.50, -0.50, -0.50, -0.50, -1.00,
+];
+
+const ROOK_TABLE: [f64; 64] = [
+    -0.50, -0.50, -0.50, 0.00, 0.00, -0.50, -0.50, -0.50, -1.00, -0.50, -0.50, -0.50, -0.50, -0.50,
+    -0.50, -1.00, -1.00, -0.50, -0.50, -0.50, -0.50, -0.50, -0.50, -1.00, -1.00, -0.50, -0.50,
+    -0.50, -0.50, -0.50, -0.50, -1.00, -1.00, -0.50, -0.50, -0.50, -0.50, -0.50, -0.50, -1.00,
+    -1.00, -0.50, -0.50, -0.50, -0.50, -0.50, -0.50, -1.00, 0.00, 1.00, 1.00, 1.00, 1.00, 1.00,
+    1.00, 0.00, -0.50, -0.50, -0.50, -0.50, -0.50, -0.50, -0.50, -0.50,
+];
+
+const QUEEN_TABLE: [f64; 64] = [
+    -1.00, -0.33, -0.33, 0.00, 0.00, -0.33, -0.33, -1.00, -0.33, 0.33, 0.67, 0.67, 0.67, 0.67,
+    0.33, -0.33, -0.33, 0.33, 0.83, 0.83, 0.83, 0.83, 0.33, -0.33, 0.00, 0.33, 0.83, 1.00, 1.00,
+    0.83, 0.33, 0.00, 0.33, 0.33, 0.83, 1.00, 1.00, 0.83, 0.33, 0.00, -0.33, 0.67, 0.83, 0.83,
+    0.83, 0.83, 0.33, -0.33, -0.33, 0.33, 0.67, 0.33, 0.33, 0.33, 0.33, -0.33, -1.00, -0.33, -0.33,
+    0.00, 0.00, -0.33, -0.33, -1.00,
+];
+
+const KING_TABLE: [f64; 64] = [
+    -0.53, -0.76, -0.76, -1.00, -1.00, -0.76, -0.76, -0.53, -0.53, -0.76, -0.76, -1.00, -1.00,
+    -0.76, -0.76, -0.53, -0.53, -0.76, -0.76, -1.00, -1.00, -0.76, -0.76, -0.53, -0.53, -0.76,
+    -0.76, -1.00, -1.00, -0.76, -0.76, -0.53, -0.29, -0.53, -0.53, -0.76, -0.76, -0.53, -0.53,
+    -0.29, -0.06, -0.29, -0.29, -0.41, -0.41, -0.29, -0.29, -0.06, 0.76, 0.76, 0.18, 0.06, 0.06,
+    0.18, 0.76, 0.76, 0.65, 1.00, 0.53, 0.18, 0.18, 0.53, 1.00, 0.65,
+];
+
+/// Evaluate the positional value of the board for white or black.
+/// Positive scores favor white, negative scores favor black.
+pub fn evaluate_position(board: &Chessboard, is_white: bool) -> f64 {
+    let mut score = 0.0;
+
+    for (piece, bitboard) in board.piece_bitboards() {
+        // Skip pieces that don't belong to the current player.
+        if !is_white && piece.is_ascii_uppercase() || is_white && piece.is_ascii_lowercase() {
+            continue;
+        }
+
+        // Select the appropriate piece-square table.
+        let table = match piece.to_ascii_uppercase() {
+            'P' => &PAWN_TABLE,
+            'N' => &KNIGHT_TABLE,
+            'B' => &BISHOP_TABLE,
+            'R' => &ROOK_TABLE,
+            'Q' => &QUEEN_TABLE,
+            'K' => &KING_TABLE,
+            _ => continue,
+        };
+
+        // Determine the score multiplier based on the player's color.
+        let multiplier = if !is_white { 1.0 } else { -1.0 };
+
+        // Calculate the score contribution for each active square.
+        for square in position::active_squares(bitboard) {
+            let table_index = if is_white {
+                square as usize
+            } else {
+                63 - square as usize // Mirror the table for black pieces.
+            };
+            score += multiplier * table[table_index];
```

Comment:
Instead of multiplying by the multiplier each time, we should do it once at the end. Something like:
```suggestion
            return if !is_white { score } else { -score };
```